File: sdks/node/src/sandboxes.js
import { toSandboxExecResult, toSandboxRecord } from "./models.js";

export class SandboxesClient {
  constructor(client) {
    this._client = client;
  }

  async create(spec) {
    const payload = await this._client.postJson("/v1/sandboxes", spec);
    return toSandboxRecord(payload);
  }

  async get(sandboxId) {
    const payload = await this._client.getJson(`/v1/sandboxes/${sandboxId}`);
    return toSandboxRecord(payload);
  }

  async list() {
    const payload = await this._client.getJson("/v1/sandboxes");
    return (payload.sandboxes ?? []).map((item) => toSandboxRecord({ sandbox: item }));
  }

  async exec(sandboxId, request) {
    const payload = await this._client.postJson(`/v1/sandboxes/${sandboxId}/exec`, request);
    return toSandboxExecResult(payload);
  }

  async stop(sandboxId) {
    const payload = await this._client.postJson(`/v1/sandboxes/${sandboxId}/stop`, {});
    return toSandboxRecord(payload);
  }

  async delete(sandboxId) {
    return this._client.deleteJson(`/v1/sandboxes/${sandboxId}`);
  }
}

File: sdks/go/sandboxes.go
package voidcontrol

type sandboxListResponse struct {
	Sandboxes []SandboxRecord `json:"sandboxes"`
}

type sandboxExecResponse struct {
	Result SandboxExecResult `json:"result"`
}

type SandboxesClient struct {
	client *Client
}

func (client *SandboxesClient) Create(spec map[string]any) (*SandboxRecord, error) {
	var response struct {
		Sandbox SandboxRecord `json:"sandbox"`
	}
	if err := client.client.postJSON("/v1/sandboxes", spec, &response); err != nil {
		return nil, err
	}
	return &response.Sandbox, nil
}

func (client *SandboxesClient) Get(sandboxID string) (*SandboxRecord, error) {
	var response struct {
		Sandbox SandboxRecord `json:"sandbox"`
	}
	if err := client.client.getJSON("/v1/sandboxes/"+sandboxID, &response); err != nil {
		return nil, err
	}
	return &response.Sandbox, nil
}

func (client *SandboxesClient) List() ([]SandboxRecord, error) {
	var response sandboxListResponse
	if err := client.client.getJSON("/v1/sandboxes", &response); err != nil {
		return nil, err
	}
	return response.Sandboxes, nil
}

func (client *SandboxesClient) Exec(sandboxID string, request map[string]any) (*SandboxExecResult, error) {
	var response sandboxExecResponse
	if err := client.client.postJSON("/v1/sandboxes/"+sandboxID+"/exec", request, &response); err != nil {
		return nil, err
	}
	return &response.Result, nil
}

func (client *SandboxesClient) Stop(sandboxID string) (*SandboxRecord, error) {
	var response struct {
		Sandbox SandboxRecord `json:"sandbox"`
	}
	if err := client.client.postJSON("/v1/sandboxes/"+sandboxID+"/stop", map[string]any{}, &response); err != nil {
		return nil, err
	}
	return &response.Sandbox, nil
}

func (client *SandboxesClient) Delete(sandboxID string) (*SandboxDeleteResult, error) {
	var response SandboxDeleteResult
	if err := client.client.deleteJSON("/v1/sandboxes/"+sandboxID, &response); err != nil {
		return nil, err
	}
	return &response, nil
}
